### ISSUE TITLE
Alternator table left in zombie state

### ISSUE THREAD

**Issue by nyh** [scylladb/scylladb#12358]
While I was running some benchmarks on Scylla 5.1.1, I came across a bug that I don't know how to reproduce, so if we don't have additional evidence of this bug we should probably close this issue :-( But the unsolved issue may still be helpful in the future for searching if a similar bug ever reproduces.

My Alternator benchmark begins by deleting and then re-creating a table. I did not intentionally run several of those in parallel, but it's possible that maybe I did. It's also possible I interrupted the CQL client while it was in the middle of the request doing one of the above.

As far as I can tell Scylla 5.1.1 runs without Raft for schema changes (I checked system.config and the "raft" experimental option was not turned on).

Anyway, after running these creation/deletion several times, I reached a situation where neither creation nor deletion works any more:

```
$ aws dynamodb delete-table --endpoint-url http://***:8080 --table-name usertable
An error occurred (ResourceNotFoundException) when calling the DeleteTable operation: Requested resource not found: Table: usertable not found
$ aws dynamodb create-table --endpoint-url http://***:8080 --table-name usertable --attribute-definitions AttributeName=p,AttributeType=S --key-schema AttributeName=p,KeyType=HASH --billing-mode PAY_PER_REQUEST --tags Key=system:write_isolation,Value=forbid_rmw
An error occurred (InternalServerError) when calling the CreateTable operation (reached max retries: 9): Internal server error: data_dictionary::no_such_column_family (Can't find a column family with UUID d34f0410-7fe6-11ed-81f9-a122491afde0)
```

Connecting with CQL, the table *does* exist, so it's not clear why DeleteTable said it doesn't:
```
cqlsh> describe table alternator_usertable.usertable

CREATE TABLE alternator_usertable.usertable (
    p text PRIMARY KEY,
    ":attrs" map<text, blob>
) WITH bloom_filter_fp_chance = 0.01
    AND caching = {'keys': 'ALL', 'rows_per_partition': 'ALL'}
    ...
```

Moreover,
```
cqlsh> select id from system_schema.tables where keyspace_name='alternator_usertable' and table_name='usertable';

 id
--------------------------------------
 db0b9b50-7fe6-11ed-81f9-a122491afde0
```

Yet the data directory has two (!?) different subdirectories with different ids, neither matching the one above:
```
usertable-8c4e03f07fb311edacc278e8ff374d6a
usertable-beb319507fb511edacc278e8ff374d6a
```

**Comment by mykaul** [comment]
Is this reproducible and specifically, with Raft?

**Comment by nyh** [comment]
No, I don't know how I reached this weird situation and didn't make an effort to try to create an automatic reproducer. I opened the issue thinking maybe we'll see it again soon and have an issue to refer to, but I'm not aware of anyone seeing this problem again since I opened the issue, and never had the time to go back to this issue to try to reproduce it myself.
